### PR TITLE
soss-ros2 xtypes support

### PR DESCRIPTION
**Pull request by lemunozm**

### CODE REVIEW

File: packages/ros2/src/MetaPublisher.cpp [hunk 1]
```diff
@@ -43,12 +44,12 @@ class MetaPublisher : public soss::TopicPublisher
...
   }
 
-  bool publish(const soss::Message& message) override final
+  bool publish(const xtypes::DynamicData &message) override final
   {
     const std::string topic_name = _topic_template.compute_string(message);
 
     const auto insertion = _publishers.insert(
-          std::make_pair(std::move(topic_name), nullptr));
+           std::make_pair(std::move(topic_name), nullptr));
```
Comment (LuisGP) [inline packages/ros2/src/MetaPublisher.cpp]:
```suggestion
          std::make_pair(std::move(topic_name), nullptr));
```

File: packages/ros2-test/integration/ros2__geometry_msgs.cpp [hunk 2]
```diff
@@ -56,57 +56,34 @@ geometry_msgs::msg::PoseStamped generate_random_pose(const int sec = 0)
...
+  to["header"]["stamp"]["sec"] = p.header.stamp.sec;
+  to["header"]["stamp"]["nanosec"] = p.header.stamp.nanosec;
+  to["header"]["frame_id"] = "map";
+  to["pose"]["position"]["x"] = p.pose.position.x;
+  to["pose"]["position"]["y"] = p.pose.position.y;
+  to["pose"]["position"]["z"] = p.pose.position.z;
+  to["pose"]["orientation"]["x"] = p.pose.orientation.x;
+  to["pose"]["orientation"]["y"] = p.pose.orientation.y;
+  to["pose"]["orientation"]["z"] = p.pose.orientation.z;
+  to["pose"]["orientation"]["w"] = p.pose.orientation.w;
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply packages/ros2-test/integration/ros2__geometry_msgs.cpp]: I thought the same... when I saw it :cry:

File: packages/ros2-test/integration/ros2__test_msgs.cpp [hunk 3]
```diff
@@ -22,67 +22,60 @@
...
+template<typename Ros2Primitives>
+Ros2Primitives generate_random_primitives(const std::size_t seed)
 {
-  using test_msgs::msg::BasicTypes;
+  using namespace test_msgs::msg;
 
   std::mt19937 rng;
   // Use consistent seeds for deterministic test results
-  rng.seed(64 + seed);
+  rng.seed(64+seed);
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply packages/ros2-test/integration/ros2__test_msgs.cpp]: I do not know why I wrote this so. Fixed!

File: packages/ros2-test/integration/ros2__test_msgs.cpp [hunk 4]
```diff
@@ -22,67 +22,60 @@
...
 
-#include <test_msgs/msg/basic_types.hpp>
-#include <test_msgs/msg/arrays.hpp>
-
-test_msgs::msg::BasicTypes generate_random_basic_types(const std::size_t seed)
+template<typename Ros2Primitives>
+Ros2Primitives generate_random_primitives(const std::size_t seed)
 {
-  using test_msgs::msg::BasicTypes;
+  using namespace test_msgs::msg;
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply packages/ros2-test/integration/ros2__test_msgs.cpp]: I tested, the entire line is unused.

File: packages/ros2-test/integration/ros2__geometry_msgs.cpp [hunk 5]
```diff
@@ -289,8 +260,14 @@ TEST_CASE("Talk between ros2 and the mock middleware", "[ros2]")
 
     CHECK(ros2_pose == received_pose);
   }
+
+
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply packages/ros2-test/integration/ros2__geometry_msgs.cpp]: Fixed

File: packages/core/src/Config.cpp [hunk 6]
```diff
@@ -569,7 +569,7 @@ bool Config::parse(const YAML::Node& config_node, const std::string& file)
                   << "[" << entry.first << "]" << std::endl;
         return false;
       }
-      m_required_types[it_remap.first].messages.insert(it_remap.second.type);
+      m_required_types[it_remap.first].services.insert(it_remap.second.type);
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply packages/core/src/Config.cpp]: Fixed! swap fields.

**Pull request merged**